pkg/resourcegenerator/networking/network_policy.go

```diff
@@ -58,20 +59,24 @@ func getPolicyTypes(ingressRules []networkingv1.NetworkPolicyIngressRule, egress
 	return policyType
 }
 
-func getEgressRules(accessPolicy *podtypes.AccessPolicy, namespace string, availableServices []corev1.Service) []networkingv1.NetworkPolicyEgressRule {
+func getEgressRules(opts NetPolOpts) []networkingv1.NetworkPolicyEgressRule {
 	var egressRules []networkingv1.NetworkPolicyEgressRule
+	accessPolicy := opts.AccessPolicy
+	namespace := opts.Namespace
+	namespaces := *opts.Namespaces
+	availableServices := *opts.RelatedServices
 
 	// Egress rules for internal peers
 	if accessPolicy == nil || availableServices == nil {
 		return egressRules
 	}
 
 	for _, outboundRule := range (*accessPolicy).Outbound.Rules {
-		if outboundRule.Namespace == "" {
+		if outboundRule.Namespace == "" && outboundRule.NamespacesByLabel == nil {
 			outboundRule.Namespace = namespace
 		}
 
-		relatedService, isApplicationAvailable := getRelatedService(availableServices, outboundRule)
+		relatedService, isApplicationAvailable := getRelatedService(availableServices, outboundRule, namespaces)
```

Comment:
maybe stupid question, but why do we need this step? wont it return the same result as `GetEgressServices`, so the same values that already exist in `availableServices`?

Reply:
Not sure. May be that can be optimised.

What I can see is that`GetEgressServices` takes an `accessPolicy`, whereas `getRelatedService` takes an `InternalRule`. Here we are working in the context of a single rule instead of the entire access policy. So it seems that this function further filters the result set into a finer subset. That may be significant.